xblock/fields.py

```diff
@@ -780,11 +780,19 @@ class String(JSONField):
     MUTABLE = False
 
     def from_json(self, value):
-        if value is None or isinstance(value, basestring):
+        if value is None:
+            return ''
+        elif isinstance(value, basestring):
             return value
         else:
             raise TypeError('Value stored in a String must be None or a string, found %s' % type(value))
 
+    def to_json(self, value):
+        if value is None:
+            return ''
+        self._warn_deprecated_outside_JSONField()
+        return value
```

Comment:
This is better, from the perspective of not causing bugs. It's a bit unfortunate that it effectively makes the semantics in OLX different than in code, since the serialized version can't actually support `None`. So, roundtripping `None` out to OLX and back turns it into `''`, unless `None` is the default value for the `String`.